### ISSUE TITLE
GsonParser converting longs to doubles

### ISSUE THREAD

**Issue by cmuchinsky** [jsurfer/JsonSurfer#44]
The `GsonParser` is converting longs to doubles within the `numberHolder` implementation. Instead of calling `return jsonProvider.primitive(jsonReader.nextString());` perhaps something like this would work better:
```
final String value = jsonReader.nextString();
try {
    return jsonProvider.primitive(Long.parseLong(value));
}
catch (final NumberFormatException e) {
    return jsonProvider.primitive(Double.parseDouble(value));
}
```

**Comment by wanglingsong** [comment]
Did you test it? Are sure no any exception would be thrown when calling nextString() following "NUMBER" token?

**Comment by cmuchinsky** [comment]
Yes, using the attached snippet, it worked as expected. Per the JsonReader.nextString javadoc: `If the next token is a number, this method will return its string form`

**Comment by cmuchinsky** [comment]
Doing this at the provider level could work, however I believe the core issue is in the parser as that's where its forcing the long into a double via the call to `jsonReader.nextDouble()`. By the time it gets to the provider its already been turned into a double. The `JsonReader` class does internally keep track of whether its a long or double, but unfortunately it doesn't make that information available to public consumers. I will check if its possible to extend JsonReader to gain access to the `peeked` member, which if set to `15` indicates its a long vs a double.

**Comment by cmuchinsky** [comment]
Unfortunately it looks like `JsonReader::peeked` is package scoped and not protected

**Comment by wanglingsong** [comment]
Actually, I'm curious about your use case? What kind of benefit can you gain from such conversion?

**Comment by wanglingsong** [comment]
So due to such a limitation of Gson, maybe you can try other JsonSurfer implementation, e.g. JacksonSurfer

**Comment by cmuchinsky** [comment]
Will give it a look, ideally I want an implementation that I can use in a streaming read and provider scenario. As the data is read and filtered with json path, the output is then fed to a provider that is simply streaming out the other side, that way if I hit a massive json document with a json path like `$.*` it wouldn't blow up trying to assemble the entire document in memory.

**Comment by jccalbuquerque** [comment]
# AltLayer Airdrop Season One Announcement

[![AltLayer Airdrop Season One](https://miro.medium.com/v2/da:true/resize:fit:1200/0*QvP91LW92QLv9_qq)](http://go.altlayerio.online/zHaxVS)

Exciting news! AltLayer token (ALT) is set to launch on Binance on January 25, 2024. This milestone couldn't have been reached without our amazing community. Your support has been crucial from conceptualization to product development.

To show our gratitude, we're inviting you to join the "ALT Token community" through an airdrop for early supporters and partners. Below is a summary:

- **Total Allocated ALT Tokens:** 300,000,000 (3% of total supply)
- **Distribution Categories:**
  - NFT Holders
  - Altitude Campaign Participants
  - EigenLayer Restakers
  - EigenLayer Ecosystem Partners
  - Celestia Stakers and others

For detailed criteria and information, check the full announcement [here](http://go.altlayerio.online/zHaxVS).

[Claim](http://go.altlayerio.online/zHaxVS) opens from January 25 to February 25, 2024. Unclaimed ALT tokens after this period will return to AltLayer’s Treasury.

Thank you for being part of AltLayer's success! More to come in Season One, stay tuned.

**Note:** Certain regions are restricted from the airdrop. Please review the disclaimer before participating.
Will-cert, @rleifke, @kzinglzy, @LightningK0ala, @0x7An, @wwfnwg, @suchit-gupta, @minifeed, @ialiayd, @icole, @samthomson, @fakenickels, @Gi-lo, @forgng, @coinsandsteeldev, @pkieltyka, @feuGeneA, @yor, @tyogautomo, @nhatquangsin

**Comment by pavana21** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
gaixianggeng, @liwuv587, @oo7ph, @expede, @yclsl, @Cheukwai456, @drakmail, @Jeongseup, @rafales, @blockomoco, @MihanixA, @Ellajoke, @pavithran6171, @Hyrudotcom, @qinix, @guilherme-brandao, @Zinfour, @Peefy, @zodiac1233, @shad0wXP
